platform/ecsv3/ecs.go

```diff
@@ -143,3 +144,29 @@ func detectNetworkMode(meta *ecsTypes.TaskResponse) (networkMode, error) {
 		return "", fmt.Errorf("unsupported NetworkMode: %v", nm)
 	}
 }
+
+func getTaskMetadata(ctx context.Context, client TaskMetadataGetter, interval time.Duration) *ecsTypes.TaskResponse {
+	var meta *ecsTypes.TaskResponse
+	var duration time.Duration
+
+loop:
```

Comment:
In first loop, this function calls `GetTaskMetadata` immediately, in second or later, this waits while interval then retry `GetTaskMetadata`. Right?

How is this?

```go
for {
    meta, err := client.GetTaskMetadata(ctx)
    if err == nil {
        return meta
    }
    select {
    case <-time.After(interval):
        // snip
    case <-ctx.Done():
        // snip
    }
}
```

Reply:
Thank you! I will adopt this code.